agent/mitm/handlers/RequestSession.ts

```diff
@@ -130,6 +130,14 @@ export default class RequestSession
     }
   }
 
+  public async haveSentResponse(context: IMitmRequestContext): Promise<void> {
+    context.setState(ResourceState.EmulationHaveSentResponse);
```

Comment:
This state appears to be redundant with the `ResourceState.End` event that occurs right after `context.requestSession.haveSentResponse` in the HttpRequestHandler